src/containerapp/azext_containerapp/tests/latest/utils.py

```diff
@@ -1,321 +1,344 @@
+
+            while managed_env["properties"]["provisioningState"].lower() == "waiting":
+                time.sleep(5)
+                managed_env = test_cls.cmd('containerapp env show -g {} -n {}'.format(rg_name, env_name)).get_output_in_json()
+    return managed_env["id"]
+
+
+def create_containerapp_env(test_cls, env_name, resource_group, location=None):
+    logs_workspace_name = test_cls.create_random_name(prefix='containerapp-env', length=24)
+    logs_workspace_id = test_cls.cmd('monitor log-analytics workspace create -g {} -n {} -l {}'.format(resource_group, logs_workspace_name, TEST_LOCATION)).get_output_in_json()["customerId"]
+    logs_workspace_key = test_cls.cmd('monitor log-analytics workspace get-shared-keys -g {} -n {}'.format(resource_group, logs_workspace_name)).get_output_in_json()["primarySharedKey"]
+
+    if location:
+        test_cls.cmd(f'containerapp env create -g {resource_group} -n {env_name} --logs-workspace-id {logs_workspace_id} --logs-workspace-key {logs_workspace_key} -l {location}')
+    else:
+        test_cls.cmd(f'containerapp env create -g {resource_group} -n {env_name} --logs-workspace-id {logs_workspace_id} --logs-workspace-key {logs_workspace_key}')
+
+    containerapp_env = test_cls.cmd('containerapp env show -g {} -n {}'.format(resource_group, env_name)).get_output_in_json()
+
+    while containerapp_env["properties"]["provisioningState"].lower() == "waiting":
+        time.sleep(5)
+        containerapp_env = test_cls.cmd('containerapp env show -g {} -n {}'.format(resource_group, env_name)).get_output_in_json()
+
+
+def create_and_verify_containerapp_up(
+            test_cls,
+            resource_group,
+            env_name = None,
+            source_path = None,
+            artifact_path = None,
+            image = None,
+            location = None,
+            ingress = None,
+            target_port = None,
+            app_name = None,
+            requires_acr_prerequisite = False):
+        # Configure the default location
+        test_cls.cmd('configure --defaults location={}'.format(TEST_LOCATION))
+
+        # Ensure that the Container App environment is created
+        if env_name is None:
+           env_name = test_cls.create_random_name(prefix='env', length=24)
+           test_cls.cmd(f'containerapp env create -g {resource_group} -n {env_name}')
+
+        if requires_acr_prerequisite:
+            # Create ACR
+            registry_name = test_cls.create_random_name(prefix='containerapp', length=24)
+            acr = test_cls.cmd('acr create -g {} -n {} --sku Basic --admin-enabled'.format(resource_group, registry_name)).get_output_in_json()
+            registry_server = acr["loginServer"]
+            acr_credentials = test_cls.cmd('acr credential show -g {} -n {}'.format(resource_group, registry_name)).get_output_in_json()
+            registry_user = acr_credentials["username"]
+            registry_pass = acr_credentials["passwords"][0]["value"]

```

Comment:
Are the values meant to be used in this function?

Reply:
Yes! This is coming from another test helper file to create an ACR instance (with some minor changes).
I think however that I was missing one reference to "registry_pass" in my code. I updated the logic to make it easier to follow and more compact in a single if.